realm/src/main/java/io/realm/Realm.java

```diff
@@ -1060,30 +1086,70 @@ public void writeEncryptedCopyTo(File destination, byte[] key) throws IOExceptio
-            Class<?> generatedClass = getProxyClass(clazz);
+            table = createTable(clazz);
+        }
 
-            Method method = initTableMethods.get(generatedClass);
-            if (method == null) {
-                try {
-                    method = generatedClass.getMethod("initTable", new Class[]{ImplicitTransaction.class});
-                } catch (NoSuchMethodException e) {
-                    throw new RealmException("Could not find the initTable() method in generated proxy class: " + APT_NOT_EXECUTED_MESSAGE);
-                }
-                initTableMethods.put(generatedClass, method);
-            }
+        long rowIndex = table.addEmptyRow();
+        return get(clazz, rowIndex);
+    }
 
-            try {
-                table = (Table) method.invoke(null, transaction);
-                tables.put(clazz, table);
-            } catch (IllegalAccessException e) {
-                throw new RealmException("Could not launch the initTable method: " + APT_NOT_EXECUTED_MESSAGE);
-            } catch (InvocationTargetException e) {
-                e.printStackTrace();
-                throw new RealmException("An exception occurred while running the initTable method: " + APT_NOT_EXECUTED_MESSAGE);
-            }
+    /**
+     * Instantiates a new object, add it to the Realm and sets field values. Only fields of types
+     * boolean, short, int, long, float, double, String, and Date are supported.
+     *
+     * @param clazz The Class of the object to create
+     * @param values Key/value pairs with field names as keys. Missing field names are silently
+     *               ignored.
+     * @return The new object
+     * @throws java.lang.IllegalArgumentException if a field name does not exist or value is of the
+     *                                            wrong type.
+     */
+    public <E extends RealmObject> E createObjectWithValues(Class<E> clazz, HashMap<String, Object> values) {
+        checkIfValid();
+        Table table = tables.get(clazz);
+        if (table == null) {
+            table = createTable(clazz);
         }
-
         long rowIndex = table.addEmptyRow();
+        for (String key : values.keySet()) {
+            long columnIndex = table.getColumnIndex(key);
+            if (columnIndex == -1) {
+                throw new IllegalArgumentException(String.format("Field '%s' does not exist in class '%s'.",
+                        key, clazz.getCanonicalName()));
+            }
+            ColumnType columnType = table.getColumnType(columnIndex);
+            Object object = values.get(key);
+            if (columnType.matchObject(object)) {
+                switch (columnType) {
+                    case BOOLEAN:
+                        table.setBoolean(columnIndex, rowIndex, ((Boolean) object).booleanValue());
+                        break;
+                    case INTEGER:
+                        table.setLong(columnIndex, rowIndex, ((Long) object).longValue());
```

Comment:
Column type of type INTEGER is matching Long, Integer, Short and Byte.
That line will cause a ClassCastException if your object is not a Long.